### PR TITLE
Makes simple slimes ignore slime people when searching for targets again

### LINKED ISSUE #12518: Slimes eat slime people
Xenobiology slimes would previously not attack slime people.

This was changed with an undocumented tweak introduced in #12398

https://github.com/ParadiseSS13/Paradise/pull/12398/files#diff-3b7ff7acf97c44b0c42afd443293685cL328

S Q U I S H

### PR DESCRIPTION
**Pull request by SteelSlayer**
## What Does This PR Do
Simple slimes will ignore slime people when searching for targets, as they used to do.

EDIT: Updated to include species `inherent_factions` and gives slime people the "slime" faction. Since simple slimes have this faction, they will never attack other simple slimes, or slime people.

## Why It's Good For The Game
Restores a gameplay mechanic that was removed in the slime refactor (#12398) as an undocumented change.

Fixes #12518

## Changelog
:cl:
add: Adds in an inherent_factions var to all species. Gives slime people the "slime" faction
fix: Slimes will ignore slime people when searching for targets again
/:cl:

### CODE REVIEW

**Comment by Mitchs98** [comment]
Yay. *squish

**Comment by Fox-McCloud** [comment]
Instead of being a ❄️ why not make it so that you use something like TG's `inherent_factions` list.

**Comment by SteelSlayer** [comment]
I'll look into it

**Comment by Mitchs98** [comment]
With any luck this'll get merged soon 👀

**Comment by Fox-McCloud** [comment]
You'll have to alter pod diona and shadow people, as well.

Then this should be ready to go.

**Comment by SteelSlayer** [comment]
Alright, just confirming this is what you want.. I looked up what factions they have on tg. For pod people, they'll get the "plants" and "vines" faction, and for shadow's, they'll get the "faithless" faction, is that right?

File: code/modules/mob/living/simple_animal/slime/life.dm [hunk 1]
```diff
@@ -314,7 +314,7 @@
 
 				for(var/mob/living/L in view(7,src))
 
-					if(isslime(L) || L.stat == DEAD) // Ignore other slimes and dead mobs
+					if(L.stat == DEAD) // Ignore dead mobs
```
Comment (Fox-McCloud) [inline code/modules/mob/living/simple_animal/slime/life.dm:317]: out of curiosity, why did you remove this?

Comment (SteelSlayer) [reply code/modules/mob/living/simple_animal/slime/life.dm:317]: Further down on line 327, the slime checks if the target mob has one of its own factions. Simple animal slimes get the "slime" faction which would mean they ignore other slimes. The check is redundant in this case.
